extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -586,7 +595,10 @@ export class AmpA4A extends AMP.BaseElement {
                         'Key failed to validate creative\'s signature',
                         keyInfo.serviceName, keyInfo.cryptoKey);
                   }
-                  return null;
+                  // Reject to ensure the some operation waits for other
+                  // possible providers to properly verify and resolve.
+                  return Promise.reject(
+                      `${keyInfo.serviceName} failed to verify`);
```

Comment:
I know that writing unit tests for async code / testing race conditions is a nightmare, but I'd be a lot more comfortable if we could have one or more tests that fails when the race condition is triggered, but passes with your change to the code here.